internal/repository/repository_test.go

```diff
@@ -0,0 +1,159 @@
+// (c) Cartesi and individual authors (see AUTHORS)
+// SPDX-License-Identifier: Apache-2.0 (see LICENSE)
+
+package repository
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/stretchr/testify/suite"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/modules/postgres"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const testTimeout = 300 * time.Second
+
+// This suite sets up a container running a postgres database
+type RepositorySuite struct {
+	suite.Suite
+	ctx      context.Context
+	cancel   context.CancelFunc
+	postgres *postgres.PostgresContainer
+	database *database
+}
+
+func (s *RepositorySuite) SetupSuite() {
+	s.ctx, s.cancel = context.WithTimeout(context.Background(), testTimeout)
+
+	var err error
+	s.postgres, err = newPostgresContainer(s.ctx)
+	s.Require().Nil(err)
+
+	endpoint, err := s.postgres.ConnectionString(s.ctx, "sslmode=disable")
+	s.Require().Nil(err)
+	RunMigrations(endpoint)
+
+	s.database, err = Connect(s.ctx, endpoint)
+	s.Require().Nil(err)
+}
+
+func (s *RepositorySuite) TearDownSuite() {
+	err := s.postgres.Terminate(s.ctx)
+	s.Nil(err)
+	s.cancel()
+}
+
+func (s *RepositorySuite) TestInputExists() {
+
+	input := Input{
+		Index:  0,
+		Status: "UNPROCESSED",
+		Blob:   common.Hex2Bytes("deadbeef"),
+	}
+
+	err := s.database.InsertInput(s.ctx, &input)
+	s.Require().Nil(err)
+
+	response, err := s.database.GetInput(s.ctx, 0)
+	s.Require().Nil(err)
+
+	s.Require().Equal(&input, response)
+}
+
+func (s *RepositorySuite) TestInputFailsDoesntExist() {
+	response, err := s.database.GetInput(s.ctx, 1)
+	s.Require().Nil(response)
+
+	s.Require().Contains(fmt.Sprintf("%v", err), "no rows in result set")
```

Comment:
It seems that `ErrorContains` is the function you wanted 😉 
```suggestion
	s.Require().ErrorContains(err, "no rows in result set")
```